paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkConnectorOptions.java

```diff
@@ -224,6 +224,18 @@ public class FlinkConnectorOptions {
                     .withDescription(
                             "If the new snapshot has not been generated when the checkpoint starts to trigger, the enumerator will block the checkpoint and wait for the new snapshot. Set the maximum waiting time to avoid infinite waiting, if timeout, the checkpoint will fail. Note that it should be set smaller than the checkpoint timeout.");
 
+    public static final ConfigOption<Boolean> LOOKUP_ASYNC =
+            ConfigOptions.key("lookup.async")
```

Comment:
Just as I mentioned in previous PR, it's a bit strange to mix the hint of `Lookup('async' = 'true')` with the `lookup.async` option.

Reply:
Yes, but it is a necessity.

Flink will choose asynchrony by default, but we do not want to provide asynchrony by default because asynchrony is meaningless.

So I introduce this option to provide sync mode by default.

Reply:
I see, since Flink's `preferAsync` [returns true except async option in hint is false](https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/utils/LookupJoinUtil.java#L482-L485), how about we could simplify the example to not include the hint of `Lookup('async' = 'true')`?

Reply:
Good point! You are right!